tests/math_utils.test.ts

```diff
@@ -85,3 +90,128 @@ describe("remap", () => {
     expect(remap(2, 0, 1, 0, 10, false)).to.equal(20);
   });
 });
+
+describe("getFrameSizeInScreenPx", () => {
+  it("returns canvas size when frame is the same aspect ratio", () => {
+    const frameSizePx = getFrameSizeInScreenPx([100, 100], [1, 1], 1);
+    expect(frameSizePx).to.deep.equal([100, 100]);
+  });
+
+  it("does not change returned size if frame resolution is higher", () => {
+    const frameResolutions: [number, number][] = [
+      [0.1, 0.1],
+      [1, 1],
+      [10, 10],
+      [100, 100],
+    ];
+    for (const frameResolution of frameResolutions) {
+      const frameSizePx = getFrameSizeInScreenPx([100, 100], frameResolution, 1);
+      expect(frameSizePx).to.deep.equal([100, 100]);
+    }
+  });
+
+  it("constrains by height when frame aspect ratio is taller than canvas", () => {
```

Comment:
I would appreciate these descriptions being clearer. maybe it's me, but "constrains by" sounds like that's going to be the smaller value? And is the frameResolution "taller than the canvas" here? if the height is 1, I think that's the same size as the canvas. The ratio is narrower than the canvas but not taller? am I understanding this correctly?